plenum/test/txn_author_agreement/test_config_req_handler_taa_utils.py

```diff
@@ -31,61 +37,101 @@ def test_is_trustee(txnPoolNodeSet, sdk_wallet_trustee, sdk_wallet_steward, sdk_
-    agreements = [(payload,
-                   ConfigReqHandler._taa_digest(payload[TXN_AUTHOR_AGREEMENT_VERSION],
-                                                payload[TXN_AUTHOR_AGREEMENT_TEXT]))
-                  for payload in agreements]
-    versions = [payload.get(TXN_AUTHOR_AGREEMENT_VERSION) for payload, _ in agreements]
+def test_update_txn_author_agreement(
+    config_req_handler: ConfigReqHandler, taa_input_data,
+    taa_expected_state_data, taa_expected_digests
+):
+    """ `update_txn_author_agreement` updates state properly """
     state = config_req_handler.state
-
-    for payload, digest in agreements:
-        config_req_handler.update_txn_author_agreement(payload)
-
-        assert config_req_handler.get_taa_digest(isCommitted=False) == \
-               state.get(ConfigReqHandler._state_path_taa_latest(), isCommitted=False)
-
-        for version in versions:
-            assert config_req_handler.get_taa_digest(version=version, isCommitted=False) == \
-                   state.get(ConfigReqHandler._state_path_taa_version(version), isCommitted=False)
-
-
-def test_multiple_update_txn_author_agreement(config_req_handler: ConfigReqHandler):
-    text_v1 = 'Some agreement'
-    agreement_v1 = {TXN_AUTHOR_AGREEMENT_VERSION: 'v1', TXN_AUTHOR_AGREEMENT_TEXT: text_v1}
-    digest_v1 = ConfigReqHandler._taa_digest('v1', text_v1).encode()
-    config_req_handler.update_txn_author_agreement(agreement_v1)
-
-    assert config_req_handler.get_taa_digest(isCommitted=False) == digest_v1
-    assert config_req_handler.get_taa_digest(version='v1', isCommitted=False) == digest_v1
-    assert config_req_handler.get_taa_digest(version='v2', isCommitted=False) is None
-
-    text_v2 = 'New agreement'
-    agreement_v2 = {TXN_AUTHOR_AGREEMENT_VERSION: 'v2', TXN_AUTHOR_AGREEMENT_TEXT: text_v2}
-    digest_v2 = ConfigReqHandler._taa_digest('v2', text_v2).encode()
-    config_req_handler.update_txn_author_agreement(agreement_v2)
-
-    assert config_req_handler.get_taa_digest(isCommitted=False) == digest_v2
-    assert config_req_handler.get_taa_digest(version='v1', isCommitted=False) == digest_v1
-    assert config_req_handler.get_taa_digest(version='v2', isCommitted=False) == digest_v2
+    written = []
+
+    def _check_state(version):
+        digest = taa_expected_digests[version]
+
+        _digest = state.get(
+            ConfigReqHandler._state_path_taa_version(version),
+            isCommitted=False
+        )
+        _data = state.get(
+            ConfigReqHandler._state_path_taa_digest(digest),
+            isCommitted=False
+        )
+
+        if version in written:
+            assert _digest == digest.encode()
+            assert (
+                config_state_serializer.deserialize(_data) ==
+                taa_expected_state_data[version]
+            )
+        else:
+            assert _digest is None
+            assert _data is None
+
+    for data in taa_input_data:
```

Comment:
This is of course a matter of taste, and not a subject for change, but I would prefer more simple and readable unit tests (closer to the initial version of the tests).